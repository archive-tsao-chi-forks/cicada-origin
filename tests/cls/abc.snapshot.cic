@def the_nat_type
@the (Trivial) -> Type
(_) => Nat

@def abc
@the [ a: Trivial, b: the_nat_type(a), c: String ]
{ a: sole, b: 2, c: "c" }

@show abc

@show abc.a
@show abc.b
@show abc.c

@def xyz
@the [
  x: String,
  y: String,
  z: [ a: Trivial, b: the_nat_type(a), c: String ]
] {
  x: "x",
  y: "y",
  z: abc,
}

@show xyz

@show xyz.z
@show xyz.z.a
@show xyz.z.b
@show xyz.z.c
