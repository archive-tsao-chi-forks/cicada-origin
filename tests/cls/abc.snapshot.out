@the [
  a: Trivial
  b: Nat
  c: String
] {
  a: sole
  b: 2
  c: "c"
}
@the Trivial sole
@the Nat 2
@the String "c"
@the [
  x: String
  y: String
  z: [
    a: Trivial
    b: Nat
    c: String
  ]
] {
  x: "x"
  y: "y"
  z: {
    a: sole
    b: 2
    c: "c"
  }
}
@the [
  a: Trivial
  b: Nat
  c: String
] {
  a: sole
  b: 2
  c: "c"
}
@the Trivial sole
@the Nat 2
@the String "c"

